quizapp.js

```diff
@@ -69,6 +82,73 @@ function onClickingShuffle(event) {
   }
 }
 function displayQuestion(item) {
-  const question = document.getElementById("quizquestions question");
-  console.log(questionArray);
+  const section = document.getElementById("quizquestions");
+  section.innerHTML = "";
+  const question = document.createElement("div");
+  section.appendChild(question);
+  const p = document.createElement("p");
+  p.innerText = item.id + " " + item.question;
+  question.appendChild(p);
+  item.options.forEach((option) => {
+    const optionBtn = document.createElement("button");
+    optionBtn.classList.add("btnList");
+    optionBtn.innerText = option.text;
+    optionBtn.dataset.correct = option.isCorrect;
+    question.appendChild(optionBtn);
+    optionBtn.addEventListener("click", (event) =>
+      checkAnswer(event, question)
+    );
+  });
+  const solutionBtn = document.createElement("button");
+  question.appendChild(solutionBtn);
+  solutionBtn.classList.add("solutionbtn");
+  solutionBtn.innerText = "Solution";
+  solutionBtn.dataset.explanation = item.explanation;
+  solutionBtn.addEventListener("click", revealSolution);
+}
+
+function revealSolution(event) {
+  const button = event.target;
+  const explanation = button.dataset.explanation;
+  const p = document.createElement("p");
+  p.innerText = explanation;
+  button.parentNode.appendChild(p);
+  if (button) {
+    button.disabled = true;
+  }
+}
+
+function checkAnswer(event, question) {
+  const btnSelected = event.target;
+  const isCorrect = btnSelected.dataset.correct === "true";
+  const buttons = question.querySelectorAll("button");
+  buttons.forEach((button) => {
+    if (button !== btnSelected) {
+      button.disabled = true;
+    }
+  });
+  if (isCorrect) {
+    btnSelected.classList.add("correct");
+  } else {
+    btnSelected.classList.add("in-correct");
+  }
+}
+function searchQuestion(event) {
+  const string = event.target.value.toLowerCase();
+  const question = questionArray.filter((item) =>
+    item.question.includes(string)
+  );
```

Comment:
You will need to refactor this to all check not just the question string but also it's contents so at least the answers but perhaps also the explanation to have a proper search.

Reply:
Thank you for your time. I have changed the code to check the options and explanation as well and if match found, it will display the question and its options.